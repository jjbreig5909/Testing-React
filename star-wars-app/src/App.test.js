import axios from "axios"
import React from "react";
// no default export, so we're importing everyting from this library
import * as rtl from "@testing-library/react";
// not importing to a variable, since this just overrides jest global variables
import "@testing-library/jest-dom/extend-expect";
import App from "./App";



jest.mock("axios", () => {
    return {
        get: jest.fn(() => Promise.resolve({
            data: {
                results: ["foo.jpg", "bar.jpg"],
                previous: "null",
                next: "next-page"
            }
        }))
    }
})



afterEach(rtl.cleanup);

test('Do buttons render?', () => {
    const { getByTestId } = rtl.render(<App />);
    getByTestId('next');
    getByTestId('prev');
});

test('Is the logo there?', () => {
    const wrapper=rtl.render(<App />);
    const element = wrapper.getByAltText(/logo/i);
    expect(element).toBeVisible();
})

test("<App /> snapshot", async () => {
    const wrapper = rtl.render(<App />)
    await wrapper.findAllByAltText(/logo/i);
    expect(wrapper.asFragment()).toMatchSnapshot()
})

test("Made API call", async () => {
    const wrapper = rtl.render(<App />);
    await wrapper.findAllByAltText(/logo/i);
    expect(axios.get).toHaveBeenCalled()
})


