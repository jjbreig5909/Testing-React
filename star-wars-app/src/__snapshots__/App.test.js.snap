// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        src="https://logos-download.com/wp-content/uploads/2016/09/Star_Wars_logo-1.png"
        width="300"
      />
    </header>
    <div>
      <div />
      <div />
      <div
        class="buttons"
      >
        <button
          data-testid="prev"
        >
          Previous
        </button>
        <button
          data-testid="next"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
